docs/proposals/service-catalog.md

```diff
@@ -0,0 +1,408 @@
+username/password be provisioned. The job, however, would need to have
+sufficient privileges to create a username/password, which means the
+administrative credentials necessary to provision the new account would need to
+be included in the chart itself, which is a security risk and presumably not a
+viable solution. Alternatively, the job would need to make an unprivileged
+request to some other service that does have sufficient privileges.
+
+TODO
+Also: 7.1, use pre-hooks to provision resources
+
+### Docker Compose
+
+Docker Compose is most like a pod spec file.  It can contain information about
+container(s) to be deployed together (similar to a pod) and the volumes,
+environment variables, and ports that the combined composition will use.
+
+Compose is similar to Helm in that it can bring up and take down a templated set
+of resources as a unit.  Obviously, Compose does not create Kubernetes resources
+as Helm does, including Services which make the service offered by the
+composition discoverable.
+
+### OpenShift Templates
+
+Openshift templates offer a feature set similar to Helm.  Templates are
+resources created as yaml files and imported as a resource of type "Template"
+into Openshift.  Templates are processed with the "oc process" command, filling
+in fields from parameter list and can even dynamically generate parameters like
+passwords, for example. Once processed, the template becomes a single yaml file
+output that can be fed to `oc create -f` to create the resources in Openshift.
+
+The use case application is the same as that of Helm.  Once resources created by
+the template, they are not tracked any further as a member of the template.  It
+is up to the users to apply labels such that all resources created by a template
+can be selected in the cluster.
+
+### Cloud Foundry Service Broker
+
+The CF Service Broker is more full featured than the previous examples, in that
+it implements a catalog of services (CCDB) provided by any number of service
+brokers and claimable by cluster users.  The cloud controller can discover
+services from service brokers that implement it. the Service Broker API
+using the "catalog" call.  Other calls that happen in response to a user
+claiming the service from the catalog are "provision", "bind", "unbind", and
+"deprovision".  The separation of the provisioning stage from the binding stage
+allows for asynchronous dynamic provisioning of a service for a particular user.
+
+This satisfies all the use cases except for the syncing of changed credentials.
+There is no mechanism for the Service Broker to initiate a deprovision or
+resync.
+
+## Prototype
+
+We have implemented a prototype to demonstrate a possible workflow for Catalogs
+in a Kubernetes cluster.  The implementation is
+[here](https://github.com/sjenning/kubernetes/tree/catalog-apiserver).
+
+This prototype allows for the following type of workflow:
+
+1. A user (the “provider” or “publisher”) creates one or more resources that
```

Comment:
Is publishing shared resources other than for service consumption an important use case?